homeassistant/components/homewizard/config_flow.py

```diff
@@ -211,29 +212,33 @@ async def async_step_reauth_confirm(
         )
 
     @staticmethod
-    async def _async_try_connect(ip_address: str) -> str | None:
+    async def _async_try_connect(ip_address: str) -> tuple[Device | None, str | None]:
```

Comment:
@emontnemery

This feels a bit awkward because it returns a `Device` OR `str` (and not even `None`). But checking the return type as caller feels even weirder. What do you think is a nice approach?

Reply:
Isolated commit: https://github.com/home-assistant/core/pull/84134/commits/45cbe8440eedc38f0e6767cf79a9d3fbd53ebadd